compiler/src/iree/compiler/API/Compiler.h

```diff
@@ -53,26 +53,6 @@ MLIR_CAPI_EXPORTED void ireeCompilerOptionsGetFlags(
 // Compiler stages.
 //===----------------------------------------------------------------------===//
 
-// Builds a pass pipeline to cleanup MHLO dialect input derived from XLA.
-// A pass pipeline of this plus ireeCompilerBuildMHLOImportPassPipeline is
-// equivalent to ireeCompilerOptionsSetInputDialectXLA as a one-shot.
-MLIR_CAPI_EXPORTED void ireeCompilerBuildXLACleanupPassPipeline(
-    MlirOpPassManager passManager);
-
-// Builds a pass pipeline to lower IREE-compatible MHLO functions and ops to
-// be a legal input to IREE. This performs the standalone work that
-// ireeCompilerOptionsSetInputDialectMHLO will do as a one-shot. Notably, this
-// requires that XLA control flow has been legalized to SCF or CFG and that
-// no tuples are in the input program.
-MLIR_CAPI_EXPORTED void ireeCompilerBuildMHLOImportPassPipeline(
-    MlirOpPassManager passManager);
-
-// Builds a pass pipeline to lower IREE-compatible TOSA function and ops to
-// be a legal input to IREE. This performs that standalone work that
-// ireeCompilerOptionsSetInputDialectTOSA will do as a one-shot.
-MLIR_CAPI_EXPORTED void ireeCompilerBuildTOSAImportPassPipeline(
-    MlirOpPassManager passManager);
```

Comment:
If we really want to keep the standalone import pass pipelines, we could add 

```c
ireeCompilerBuildImportPassPipeline(InputDialectOptions inputDialectOptions, MlirOpPassManager passManager)
```

with a trimmed down version of https://github.com/iree-org/iree/blob/ca258bcd0dea4d22b4dddc9ef05382d6bd61afd3/compiler/src/iree/compiler/API/Compiler.cpp#L145-L157

calling a new function in https://github.com/iree-org/iree/blob/ca258bcd0dea4d22b4dddc9ef05382d6bd61afd3/compiler/src/iree/compiler/Pipelines/Pipelines.h#L31-L39

then associated pybind definitions.

WDYT, @stellaraccident / @benvanik?

Reply:
Not sure if anyone was using it so can't say whether it was worth it - maybe it was useful for getting post-tosa/mhlo inputs for reproducers? If no immediate uses seems fine to drop for now (smaller surface area).

Reply:
I'm fine with it. This predated the flags plumb through and everything, so once upon a time, this was the only way to get at this.